### PR TITLE
Support net7.0

### PR DESCRIPTION
**Pull request by hansmbakker**
This PR adds net7.0 to the TargetTrameworks of Tmds.Dbus.Tool to fix https://github.com/tmds/Tmds.DBus/issues/176

It also updates the GitHub Workflow and the tests projects.

### CODE REVIEW

**Comment by hansmbakker** [comment]
When testing I noticed that the error simply started indicating that the tool supports net7.0 but still showed up as an error:

```
error NU1202:
Package Tmds.DBus.Tool 0.11.1 is not compatible with netcoreapp2.2 (.NETCoreApp,Version=v2.2). Package Tmds.DBus.Tool 0.11.1 supports: 
 - net6.0 (.NETCoreApp,Version=v6.0) / any 
 - net7.0 (.NETCoreApp,Version=v7.0) / any 
 - netcoreapp3.1 (.NETCoreApp,Version=v3.1) / any 

Invalid project-package combination for Tmds.DBus.Tool 0.11.1. DotnetToolReference project style can only contain references of the DotnetTool type
```

**Comment by tmds** [comment]
> netcoreapp2.2 (.NETCoreApp,Version=v2.2)

I think your system only had .NET Core 2.2 installed when you tested this.

File: src/Tmds.DBus.Tool/Tmds.DBus.Tool.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1;net6.0;net7.0</TargetFrameworks>
```
Comment (tmds) [inline src/Tmds.DBus.Tool/Tmds.DBus.Tool.csproj:5]: .NET Core 3.1 is EOL. Please remove it here.

**Comment by tmds** [comment]
@hansmbakker thanks for the PR. Can you remove `netcoreapp3.1` from the Tool project. Looks good otherwise.

**Comment by tmds** [comment]
> Can you remove netcoreapp3.1 from the Tool project?

I've applied this change.

**Comment by tmds** [comment]
CI looks good.
Thanks!

**Comment by hansmbakker** [comment]
Hi, thank you for picking this up and completing it!

**Pull request merged**